scripts/analysis.py

```diff
@@ -0,0 +1,378 @@
+
+    return set(wasteid)
+
+
+def exec_string(array, search, whatwant):
+    """ Generates sqlite query command to select things an
+        inner join between resources and transactions.
+
+    Parmaters
+    ---------
+    array: array
+        array of criteria that generates command
+    search: str
+        where [search]
+        criteria for your search
+    whatwant: str
+        select [whatwant]
+        column (set of values) you want out.
+
+    Returns
+    -------
+    exec_str: str
+        sqlite query command.
+    """
+
+    exec_str = ('select ' + whatwant + ' from resources inner join transactions\
+                on transactions.resourceid = resources.resourceid where '
+                + str(search) + ' = ' + str(array[0]))
+
+    for ar in array[1:]:
+        exec_str += ' or ' + str(ar)
+
+    return exec_str
+
+
+def get_sum(array, column_index):
+    """ Returns sum of a column in an array
+
+    Paramters:
+    ---------
+    array: array
+        array that contains a column with numbers
+    column_index: int
+        index for the column to be summed
+    """
+    sum = 0
+    for ar in array:
+        sum += ar[column_index]
+
+    return sum
+
+
+
+def isotope_calc(wasteid_array, snf_inventory, cursor):
+    """ Calculates isotope mass using mass fraction in compositions table.
+
+        Fetches all from compositions table.
+        Compositions table has the following format:
+            SimId / QualId / NucId / MassFrac
+        Then sees if the qualid matches, and if it multiplies
+        the mass fraction by the snf_inventory.
+
+    Prameters
+    ---------
+    wasteid_array: array
+        array of qualid of wastes
+    snf_inventory: float
+        total mass of snf
+    cursor: cursor
+        cursor for sqlite3
+
+    Returns
+    -------
+    nuclide_inven: array
+        inventory of individual nuclides.
+    """
+
+    # Get compositions of different commodities
+    # SimId / QualId / NucId / MassFrac
+    cur = cursor
+    comp = cur.execute('select * from compositions').fetchall()
+    total_snf_mass = get_sum(snf_inventory, 0)
+
+    nuclide_inven = 'total snf inventory = ' + str(total_snf_mass) + 'kg \n'
+    nuclides = []
+    mass_of_nuclides = []
+    # if the 'qualid's match,
+    # the nuclide quantity and calculated and displayed.
+    for isotope in comp:
+        for num in snf_inventory:
+            # num[1] = snf inventory qualid 
+            # isotope[1] = compositions qualid 
+            if num[1] == isotope[1]:
+                # num[0] = total mass of one composition
+                # isotope[3] = mass fraction
+                nuclide_quantity = num[0] * isotope[3]
+                #isotope[2] = nucid
+                nuclide_name = isotope[2]
+                nuclides.append(nuclide_name)
+                mass_of_nuclides.append(nuclide_quantity)
+
+    nuclide_set = set(nuclides)
+    mass_dict = collections.OrderedDict({})
+
+    for nuclide in nuclide_set:
```

Comment:
loop should be a separate function that sums the nuclides in a set and returns the dictionary. 

Anyway, did you know that SQL can do filtered summations? Happy to have it happen in python, but you may get better performance from sql.